app/Http/Controllers/PreferenceController.php

```diff
@@ -24,10 +25,26 @@ public function update(Request $request)
             collect($request->only(Preferences::getValidKeys()))->filter()
         );
 
-        if (auth()->user() && $request->filled('track')) {
-            auth()->user()->track_id = $request->input('track');
+        $user = auth()->user();
 
-            auth()->user()->save();
+        if ($user) {
+            if ($request->filled('track')) {
+                $user->track_id = $request->input('track');
+            }
+
+            $request->validate([
+                'is_subscriber' => 'nullable|boolean',
+            ]);
+
+            $user->is_subscriber = $request->has('digest-subscriber');
```

Comment:
I think `track_id` and `digest-subscriber` can be added to the `validate` check before updating.